blocks/WebHook.mon

```diff
@@ -0,0 +1,169 @@
+/*
+ * $Copyright (c) 2020-2021 Software AG, Darmstadt, Germany and/or Software AG USA Inc., Reston, VA, USA, and/or its subsidiaries and/or its affiliates and/or their licensors.$
+ * Use, reproduction, transfer, publication or disclosure is prohibited except as specifically provided for in your License Agreement with Software AG
+ */
+
+package apamax.analyticsbuilder.blocks;
+
+using apama.analyticsbuilder.BlockBase;
+using apama.analyticsbuilder.Activation;
+using apama.analyticsbuilder.ABConstants;
+using apama.analyticsbuilder.L10N;
+using apama.analyticsbuilder.Value;
+
+using com.softwareag.connectivity.httpclient.HttpTransport;
+using com.softwareag.connectivity.httpclient.RequestType;
+using com.softwareag.connectivity.httpclient.Request;
+using com.softwareag.connectivity.httpclient.Response;
+
+
+/**
+ * Event definition of the parameters for the HTTP Webhook block.
+ */
+event HTTPWebhook_$Parameters {
+	
+	/**
+	 * Host.
+	 *
+	 * A valid host name or IP address.
+	 */
+	string host;
+	
+	/**
+	 * Path.
+	 *
+	 * A path component, consisting of a sequence of path segments separated by a slash (/). A path is always defined for a URI, 
+	 * though the defined path may be empty.
+	 */
+	 string path;
+	
+	/**
+	* Port.
+	*
+	* The host port number.
+	*/
+	integer port;
+	
+	/**
+	* Use HTTPS.
+	*
+	* If selected, block will use Transport-level security to transfer data over the network.  Certificate checking is not enabled.
+	*/
+	boolean tlsEnabled;
+	
+	/**
+	* "Authorization" header value.
+	*
+	* If set, block will set the "Authorization" header of the requests to the provided value.
+	*/
+	optional<string> authorizationHeader;
+	
+	/**Default value for tlsEnabled.*/
+	constant boolean $DEFAULT_tlsEnabled := false;
+
+	/** Validate that the values for all the parameters have been provided. */
+	action $validate() {
+		BlockBase.throwsOnEmpty(host, "host", self);
+		if port < 0 and port > 65535 {
+			throw L10N.getLocalizedException("HTTPWebhook_unexpected_port_value", [<any> port]);
+		}
+	}
+}
+
+/**
+ * HTTP Webhook

```

Comment:
I'd suggest 'Invoke Webhook'  - (we could also want a 'Receive Webhook' as an input block or similar)